### PR TITLE
refine downloader to fix remote load slowly bug

### PR DESCRIPTION
**Pull request by novain1102**
Re: #https://forum.cocos.org/t/topic/142426/5

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/network/DownloaderImpl-apple.mm [hunk 1]
```diff
@@ -106,7 +105,11 @@ - (void)doDestroy;
     DeclareDownloaderImplVar;
     if (task->storagePath.length()) {
 #if CC_PLATFORM == CC_PLATFORM_IOS
-        if (impl.hasUnfinishedTask == YES) {
+        NSString *requesetURL = [NSString stringWithFormat:@"%s", task->requestURL.c_str()];
+        NSString *savaPath = [NSString stringWithFormat:@"%s", (cc::FileUtils::getInstance()->getWritablePath() + "resumeData.plist").c_str()];
+        NSDictionary *dict = [NSDictionary dictionaryWithContentsOfFile:savaPath];
+        NSString *hasResumData = [dict objectForKey:requesetURL];
```
Comment (minggo) [inline native/cocos/network/DownloaderImpl-apple.mm]: hasResumData -> hasResumeData?

File: native/cocos/network/DownloaderImpl-apple.mm [hunk 2]
```diff
@@ -656,7 +663,12 @@ - (void)URLSession:(NSURLSession *)session downloadTask:(NSURLSessionDownloadTas
         }
         ccstd::vector<unsigned char> buf; // just a placeholder
         _outer->onTaskFinish(*[wrapper get], errorCode, errorCodeInternal, errorString, buf);
-        self.hasUnfinishedTask = NO;
+
+        NSMutableDictionary *dict = [NSMutableDictionary dictionary];
+        NSString *requesetURL = [NSString stringWithFormat:@"%s", [wrapper get]->requestURL.c_str()];
+        [dict removeObjectForKey:requesetURL];
+        NSString *savaPath = [NSString stringWithFormat:@"%s", (cc::FileUtils::getInstance()->getWritablePath() + "resumeData.plist").c_str()];
+        [dict writeToFile:savaPath atomically:YES];
```
Comment (minggo) [inline native/cocos/network/DownloaderImpl-apple.mm:671]: Duplicate codes. Can implement it in a function?

Comment (novain1102) [reply native/cocos/network/DownloaderImpl-apple.mm:671]: Slightly duplicate but with different purposes, I don't think it is necessary to reuse

File: native/cocos/network/DownloaderImpl-apple.mm (hunk 1 above)
Comment (novain1102) [reply native/cocos/network/DownloaderImpl-apple.mm]: fixed.

**Review by minggo — APPROVED** [review]

**Pull request merged**